src/libasr/asr_utils.cpp

```diff
@@ -713,6 +714,68 @@ ASR::asr_t* symbol_resolve_external_generic_procedure_without_eval(
     }
 }
 
+ASR::asr_t* make_ImplicitCast_t_value(Allocator &al, const Location &a_loc, ASR::expr_t* a_arg, ASR::cast_kindType a_kind, ASR::ttype_t* a_type, ASR::expr_t* value) {
+    ASR::ImplicitCast_t *n = (ASR::ImplicitCast_t *)ASR::make_ImplicitCast_t(
+        al, a_loc, a_arg, a_kind, a_type, value);
```

Comment:
Instead of creating this ahead of time, you should create it at the very end. The reason is that you first create the node, and then override some element of it, which can potentially be inefficient. We want to compute everything we need ahead of time, and then create the node.

Reply:
You are correct. I will make the changes.